raster-foundry/raster-foundry
# Migrate API spec publishing back to core repository

## Overview

Add support for linting and publishing the project API specifications after migrating the specifications themselves to this repository. Also, ensure that the Swagger CLI (used for linting) is up-to-date.

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~
~- [ ] Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
~- [ ] Any new SQL strings have tests~

### Notes

The [`prediction-spec.yml`](https://github.com/raster-foundry/raster-foundry-api-spec/blob/develop/spec/prediction-spec.yml) is still a member of the API spec repository, but I did not consider it eligible for migration because the API source code exists in another repository.

I opened https://github.com/azavea/raster-foundry-platform/issues/670 to track dealing with this, plus the eventual deprecation of the `raster-foundry-api-spec` repository.

## Testing Instructions

- Execute `test` and ensure that the API specs get validated by `swagger-cli`
- Execute `cipublish` and ensure that the local API specs get published to `rasterfoundry-staging-docs-site-us-east-1/spec`

```console
$ aws s3 ls s3://rasterfoundry-staging-docs-site-us-east-1/spec/
                           PRE 1.13.1/
                           PRE 1.17.0/
                           PRE 1.18.1/
2019-03-22 18:17:49      11738 prediction-spec.yml
2019-04-01 18:11:05     211502 spec.yml
2019-04-01 18:11:05      15546 tile-spec.yml
```


- Review the output of the [`test/deploy-api-spec`](http://jenkins.staging.rasterfoundry.com/job/Raster%20Foundry/job/raster-foundry/job/test%252Fdeploy-api-spec/) branch on Jenkins
- Review the output of the [Raster Foundry Release Pipeline](http://jenkins.staging.rasterfoundry.com/job/Raster%20Foundry%20Release%20Pipelines/job/Raster%20Foundry%20Release%20Pipeline/91/) on Jenkins

Fixes https://github.com/azavea/raster-foundry-platform/issues/649
